subprojects/docs/src/samples/user-input/publishing-credentials/README.adoc

```diff
@@ -0,0 +1,55 @@
+NOTE: You can open this sample inside an IDE using the https://www.jetbrains.com/help/idea/gradle.html#gradle_import_project_start[IntelliJ native importer] or https://projects.eclipse.org/projects/tools.buildship[Eclipse Buildship].
+
+This sample shows how credentials can be passed to a maven publish task using link:{userManualPath}/build_environment.html#sec:project_properties[project properties].
+This approach allows to externalize the sensitive configuration from the project source code and inject it only when needed.
+
+To demonstrate the concept, we will be using a stubbed maven HTTP repository that accepts the following hardcoded credentials:
+====
+include::sample[dir="groovy",files="buildSrc/src/main/java/com/example/MavenRepositoryStub.java[tags=credentials]"]
+====
+
+This project has some sample Java code to be compiled and published as a Java library.
+Gradle build file registers a publication to a maven repository that requires authentication.
+Authentication credentials are declared to be provided by Gradle properties:
+====
+include::sample[dir="groovy",files="build.gradle[tags=publication]"]
+include::sample[dir="kotlin",files="build.gradle.kts[tags=publication]"]
+====
+
+Credentials can be passed to the publish task in multiple ways now:
+
+* via command-line properties:
+=====
+----
+$ ./gradlew publish -PmavenUser=secret-user -PmavenPassword=secret-password
+----
+=====
+* via environment variables:
+=====
+----
+$ ORG_GRADLE_PROJECT_mavenUser=secret-user ORG_GRADLE_PROJECT_mavenPassword=secret-password ./gradlew publish
+----
+=====
+* by setting the properties in `gradle.properties` file:
+=====
+----
+mavenUser=secret-user
+mavenPassword=secret-password
```

Comment:
I did not add `gradle.properties` to the sample sources in order for the sanityCheck to provide test coverage and assert that just running `./gradlew` does not evaluate any missing properties eagerly and does not fail.
I think the same should be done with https://github.com/gradle/gradle/pull/13006